synapse/storage/databases/main/devices.py

```diff
@@ -1670,29 +1669,27 @@ def _add_device_outbound_room_poke_txn(
         stream_ids: List[str],
         context: Dict[str, str],
     ) -> None:
-        next_stream_id = iter(stream_ids)
-
         self.db_pool.simple_insert_many_txn(
             txn,
             table="device_lists_changes_in_room",
             keys=(
-                "stream_id",
-                "room_id",
                 "user_id",
                 "device_id",
+                "room_id",
+                "stream_id",
                 "converted_to_destinations",
                 "opentracing_context",
             ),
             values=[
                 (
-                    next(next_stream_id),
-                    room_id,
                     user_id,
                     device_id,
+                    room_id,
+                    stream_id,
                     True,  # As we're updating `device_lists_outbound_pokes` at the same time.
                     json_encoder.encode(context),
                 )
                 for room_id in room_ids
-                for device_id in device_ids
+                for device_id, stream_id in zip(device_ids, stream_ids)
```

Comment:
can we stick some comments in here, and in the `delta` file, about how the stream_ids work? It seems like we get a different stream id for each device id, but there might be multiple rooms for each stream id?